homeassistant/components/synology_dsm/camera.py

```diff
@@ -56,14 +56,14 @@ class SynoDSMCamera(SynologyDSMBaseEntity, Camera):
 
     def __init__(
         self, api: SynoApi, coordinator: DataUpdateCoordinator, camera_id: int
-    ):
+    ) -> None:
         """Initialize a Synology camera."""
         super().__init__(
             api,
             f"{SynoSurveillanceStation.CAMERA_API_KEY}:{camera_id}",
             {
-                ENTITY_NAME: coordinator.data["cameras"][camera_id].name,
-                ENTITY_ENABLE: coordinator.data["cameras"][camera_id].is_enabled,
+                ENTITY_NAME: coordinator.data["cameras"][camera_id].name,  # type: ignore[index]
+                ENTITY_ENABLE: coordinator.data["cameras"][camera_id].is_enabled,  # type: ignore[index]
```

Comment:
`coordinator.data` can be `None`. Also I think `camera_id` may not exist in `coordinator.data["cameras"]`

Reply:
this should already be considered by this (_entity is not loaded, when camera_id not exist in coordinator.data["cameras"]_)
https://github.com/home-assistant/core/blob/c778e6faf386a8a585c6d180fbf6df19d9cc2cd8/homeassistant/components/synology_dsm/camera.py#L48-L51

Reply:
Right, just add `assert coordinator.data is not None` in the beginning.